From 0c7915ad95baa537fa84b872ec3af9101f45d2e4 Mon Sep 17 00:00:00 2001
From: Tiejun Chen <tiejun.chen@windriver.com>
Date: Wed, 5 Dec 2012 15:38:31 +0800
Subject: [PATCH 6/7] net/dpa: fix kdump hang

When kdump runs, the second kernel always gets a failure when qman_init_fq
is called, but the ethernet can still work well when we skip the failure.
So return value judgement is commented off here and kdump can work well now

Signed-off-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
index 61d52e1..00eb051 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
@@ -625,12 +625,14 @@ _dpa_fq_alloc(struct list_head *list, struct dpa_fq *dpa_fq)
 		};
 
 		_errno = qman_init_fq(fq, QMAN_INITFQ_FLAG_SCHED, &initfq);
+#ifndef CONFIG_CRASH_DUMP
 		if (_errno < 0) {
 			dpaa_eth_err(dev, "qman_init_fq(%u) = %d\n",
 					qman_fq_fqid(fq), _errno);
 			qman_destroy_fq(fq, 0);
 			return _errno;
 		}
+#endif
 	}
 
 	dpa_fq->fqid = qman_fq_fqid(fq);
-- 
1.7.9.7

